RealmSwift/List.swift

```diff
@@ -615,23 +615,33 @@ extension List: MutableCollection {
             remove(at: offset)
         }
     }
-
     /// :nodoc:
     public func move(fromOffsets offsets: IndexSet, toOffset destination: Int) {
-        var tmp = [Element]()
-        for offset in offsets {
-            tmp.append(self[offset])
-        }
-        insert(contentsOf: tmp, at: destination)
-        for offset in offsets.reversed() {
-            var o = offset
-            if o >= destination {
-                o += tmp.count
+        if Element.self is EmbeddedObject.Type {
+            for offset in offsets {
```

Comment:
Is there some reason not to use this code path for non-embedded objects as well?

Reply:
It seemed like the way for non-embedded objects was "more correct," and that this was a workaround, but I can check if that actually matters

Reply:
I've added a test for this and collapsed the conditional.